llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -1866,10 +1866,19 @@ bool SimplifyCFGOpt::hoistCommonCodeFromSuccessors(Instruction *TI,
   // If either of the blocks has it's address taken, then we can't do this fold,
   // because the code we'd hoist would no longer run when we jump into the block
   // by it's address.
-  for (auto *Succ : successors(BB))
-    if (Succ->hasAddressTaken() || !Succ->getSinglePredecessor())
+  for (auto *Succ : successors(BB)) {
+    if (Succ->hasAddressTaken())
       return false;
-
+    if (Succ->getSinglePredecessor())
```

Comment:
Moreover, should us relax this condition to support mutl-cases destination in `switch`?

```suggestion
    if (Succ->getUniqPredecessor())
```

Reply:
@nikic, what do you think of such a case?